es5-shim.js

```diff
@@ -56,9 +56,17 @@ var array_push = ArrayPrototype.push;
 var array_unshift = ArrayPrototype.unshift;
 var array_concat = ArrayPrototype.concat;
 var call = FunctionPrototype.call;
+var apply = FunctionPrototype.apply;
 var max = Math.max;
 var min = Math.min;
 
+// To prevent recursion if `Function#call` or `Function#apply` are patched.
```

Comment:
This actually needs to be moved to happen on when patch is applied. I will fix it an rebase later.